backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/CPUPolicyUnit.java

```diff
@@ -47,14 +57,22 @@ public CPUPolicyUnit(PolicyUnit policyUnit,
                 continue;
             }
             Integer cores = SlaValidator.getEffectiveCpuCores(vds, context.getCluster().getCountThreadsAsCores());
-            if (cores != null && (VmCpuCountHelper.isDynamicCpuTopologySet(vm) ?
-                    vm.getCurrentNumOfCpus(false) : vm.getNumOfCpus(false)) > cores) {
-                messages.addMessage(vds.getId(), EngineMessage.VAR__DETAIL__NOT_ENOUGH_CORES.toString());
-                log.debug("Host '{}' has less cores ({}) than vm cores ({})",
-                        vds.getName(),
-                        cores,
-                        VmCpuCountHelper.getDynamicNumOfCpu(vm));
-                continue;
+            // takenCpus are CPUs (threads), we should consider switching it to cores when necessary.
+            takenCpus = context.getCluster().getCountThreadsAsCores() ? takenCpus : (int) Math.ceil(takenCpus / vds.getCpuSockets());
+            if (cores != null) {
+                int futureCpus = context.getCluster().getCountThreadsAsCores() ? cores - vds.getVmsCoresCount() :
```

Comment:
I am not sure if I should consider the "going to be taken" pCPU as well here.
Meaning, futureCpus is the above, just:
`futureCpus - numOfCpus`.